### ISSUE TITLE
Compatible with 3.5mm IR adapters?

### ISSUE THREAD

**Issue by JohnVeness** [shannah/DIYRemote#3]
Hi. Is this app compatible with (or could it be made compatible with) IR emitters that plug into a 3.5mm headphone socket, such as are available cheaply from ebay?

**Comment by shannah** [comment]
Can you send me a sample ad or spec-sheet for one of these.  I just rely on the built-in Android API for IR reading, so it depends on whether those things can be controlled that way - and if not, how they can be controlled/detected.

**Comment by JohnVeness** [comment]
Here's one on ebay: https://www.ebay.co.uk/itm/142879507527 . As far as I know they somehow turn audio into IR (the instructions mention to turn up the phone volume after connecting)! I'm not sure exactly how that works, but they are tiny, so there can't be much electronics in there.

**Comment by shannah** [comment]
I don't think this would work out of the box (but don't know for sure).  Neat idea, and a cheap way to make this applicable to any device (not just supported Android devices).  I'd need to see the SDK/API to know what would be involved in making compatibility.  Might be a simple matter - or might be more involved.   If you happen to see an SDK/API posted that I can look at, I can say more.

**Comment by JohnVeness** [comment]
I've been reading up more on these devices (there are a number of them, including videos showing how to make your own) and they are literally a couple of LEDs connected directly to the left and right outputs from the headphones. A correctly formatted audio file causes the LEDs to flash correctly. ZaZaremote appears to be the app of choice.

Problems appear to compatibility issues with some phones, and limited range due to low power of the headphone socket. Apparently some designs incorporate a battery to help with the latter problem, at the expense of added bulkiness.

I have no idea about an API or SDK, sorry.

**Comment by numeropi** [comment]
Hi, allow me to jump in, I am also a developer(python, not java). today I rescued my old IRDroid (open source hardware) 
![image](https://user-images.githubusercontent.com/56868560/103143809-3e9d0180-471e-11eb-8b52-994249fc7a18.png)

I was glad to see that it works with the irplus app for some TV's of mine. Sadly that app is private source and not reliable. Then researching I found out about DIYRemote and I loved the great idea of creating html pages to create each different device remote, which makes it very easy for non techs to tweak their remotes and see the codes inside.

I'd also like to see DIYRemote to support jack 3.5mm devices like mine (like irplus does), and I think you can find the android java code in here: https://github.com/Irdroid/Irdroid , which is the app that came when I bought my device.
